### PR TITLE
Adding permute_copy operator kernel optimization

### PR DESCRIPTION
**Pull request by Rushi-cad**
Integrating 32 bit version permute/transpose kernel

### CODE REVIEW

File: backends/cadence/hifi/kernels/kernels.h [hunk 1]
```diff
@@ -76,6 +76,14 @@ extern "C" WORD32 xa_nn_elm_where_broadcast_4D_f32xf32_f32(FLOAT32 * __restrict_
                                 const WORD32 *const p_inp2_shape,
                                 const unsigned char *__restrict__ p_condition,
                                 const WORD32 *const p_condition_shape);
+                                
+extern "C" WORD32 xa_nn_transpose_32_32(WORD32 * __restrict__ p_out,
```
Comment (dijopaul) [inline backends/cadence/hifi/kernels/kernels.h]: lint applied ?

File: backends/cadence/hifi/operators/CMakeLists.txt [hunk 2]
```diff
@@ -24,6 +24,7 @@ set(_aten_ops__srcs
     "${EXECUTORCH_ROOT}/backends/cadence/hifi/operators/op_div.cpp"
     "${EXECUTORCH_ROOT}/backends/cadence/hifi/operators/op_mean.cpp"
     "${EXECUTORCH_ROOT}/backends/cadence/hifi/operators/op_mul.cpp"
+    "${EXECUTORCH_ROOT}/backends/cadence/hifi/operators/op_permute_copy.cpp"
```
Comment (dijopaul) [inline backends/cadence/hifi/operators/CMakeLists.txt:39]: format issue

File: backends/cadence/hifi/operators/op_permute_copy.cpp [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+      xa_nn_transpose_8_8(
+          p_out,
+          p_out_shape,
+          p_inp,
+          p_inp_shape,
+          p_permute_vec,
+          num_out_dims,
+          num_inp_dims);
+
+      return out;
```
Comment (dijopaul) [inline backends/cadence/hifi/operators/op_permute_copy.cpp]: can have a common return

Comment (Rushi-cad) [reply backends/cadence/hifi/operators/op_permute_copy.cpp]: This is changed in the latest push

**Pull request merged**